commands/volumes/brickutils.go

```diff
@@ -0,0 +1,116 @@
+package volumecommands
+
+import (
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/gluster/glusterd2/brick"
+	"github.com/gluster/glusterd2/daemon"
+	"github.com/gluster/glusterd2/peer"
+	"github.com/gluster/glusterd2/utils"
+
+	"github.com/pborman/uuid"
+)
+
+// BrickStartMaxRetries represents maximum no. of attempts that will be made
+// to start brick processes in case of port clashes.
+const BrickStartMaxRetries = 3
+
+// Until https://review.gluster.org/#/c/16200/ gets into a release.
+// And this is fully safe too as no other well-known errno exists after 132
+const anotherEADDRINUSE = syscall.Errno(0x9E) // 158
+
+func errorContainsErrno(err error, errno syscall.Errno) bool {
+	if exiterr, ok := err.(*exec.ExitError); ok {
```

Comment:
No, sorry, my comment wasn't clear enough. It doesn't mean "nest it in the code", it means, inline it within itself so that it returns early. Eg:

```
func errorContainsErrno(err error, errno syscall.Errno) bool {
	exiterr, ok := err.(*exec.ExitError)
	if !ok {
		return false
	}
	status, ok := exiterr.Sys().(syscall.WaitStatus)
	if !ok {
		return false
	}
	if status.ExitStatus() != int(errno) {
		return false
	}
	return true
}

```